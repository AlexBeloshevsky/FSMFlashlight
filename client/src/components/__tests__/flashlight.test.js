import { render, screen, cleanup, waitFor } from "@testing-library/react";
import FlashLight from "../flashlight";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";

afterEach(() => {
  cleanup();
});

test("should render flashlight component", () => {
  render(<FlashLight />);
  const flashlightElement = screen.getByTestId("flashlight-1");
  expect(flashlightElement).toBeInTheDocument();
});

test("should have off state", async () => {
  render(<FlashLight />);
  await waitFor(() => {
    expect(screen.getByTestId("flashlight-2").innerHTML).toBe("off");
  });
});

test("should have on state", async () => {
  render(<FlashLight />);
  const button = screen.getByTestId("flashlight-3");
  userEvent.click(button);
  const curState = await screen.findByTestId("flashlight-2");
  expect(curState.innerHTML).toBe("on");
});

test("should have blink state", async () => {
  render(<FlashLight />);
  const button = screen.getByTestId("flashlight-3");
  userEvent.click(button);
  userEvent.click(button);
  const curState = await screen.findByTestId("flashlight-2");
  expect(curState.innerHTML).toBe("blink");
});

test("matches snapshot", () => {
  const tree = renderer.create(<FlashLight />).toJSON();
  expect(tree).toMatchSnapshot();
});
