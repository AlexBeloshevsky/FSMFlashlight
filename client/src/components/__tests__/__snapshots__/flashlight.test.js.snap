// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="flashlight-1"
>
  <h1>
    A FSM flashlight
  </h1>
  <p>
    Currently the flashlight is in the following state: 
    <span
      data-testid="flashlight-2"
    />
  </p>
  <div
    className="sc-bcXHqe LLHhx"
  >
    <div
      className="sc-gswNZR bnpZYs"
    >
      <button
        data-testid="flashlight-3"
        onClick={[Function]}
      >
        Change state
      </button>
    </div>
    <div
      className="sc-dkrFOg gWYVjy"
    />
    <div
      className="sc-hLBbgP whWOl"
    />
  </div>
</div>
`;
